nilearn/tests/test_cache_mixin.py

```diff
@@ -76,11 +76,14 @@ def test__safe_cache_flush():
 
 def test_cache_memory_level():
     temp_dir = tempfile.mkdtemp()
-    job_glob = os.path.join(temp_dir, 'joblib', '*')
+    job_glob = os.path.join(temp_dir, 'joblib', 'nilearn', 'tests',
+                            'test_cache_mixin', 'f', '*')
     mem = Memory(cachedir=temp_dir, verbose=0)
-    cache_mixin.cache(f, mem)(2)
-    assert_true(len(glob.glob(job_glob)) == 0)
     cache_mixin.cache(f, mem, func_memory_level=2, memory_level=1)(2)
     assert_true(len(glob.glob(job_glob)) == 0)
+    cache_mixin.cache(f, Memory(cachedir=None))(2)
+    assert_true(len(glob.glob(job_glob)) == 0)
```

Comment:
I'm afraid to fix stuff since people tell me that it should be part of another PR :P.